src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,693 @@
+        // Accrue any performance fees based on the increase in available assets before
+        // the liquidity asset from the redeem is reserved for the ARM withdrawal queue
+        _accruePerformanceFee();
+
+        // Calculate the amount of assets to transfer to the redeemer
+        assets = convertToAssets(shares);
+
+        requestId = nextWithdrawalIndex;
+        uint128 queued = SafeCast.toUint128(withdrawsQueued + assets);
+        uint40 claimTimestamp = uint40(block.timestamp + CLAIM_DELAY);
+
+        // Store the next withdrawal request
+        nextWithdrawalIndex = SafeCast.toUint128(requestId + 1);
+        // Store the updated queued amount which reserves WETH in the withdrawal queue
+        withdrawsQueued = queued;
+        // Store requests
+        withdrawalRequests[requestId] = WithdrawalRequest({
+            withdrawer: msg.sender,
+            claimed: false,
+            claimTimestamp: claimTimestamp,
+            assets: SafeCast.toUint128(assets),
+            queued: queued
+        });
+
+        // burn redeemer's shares
+        _burn(msg.sender, shares);
+
+        // Remove the redeemed assets from the last available assets
+        lastAvailableAssets = SafeCast.toUint128(lastAvailableAssets - assets);
+
+        emit RedeemRequested(msg.sender, requestId, assets, queued, claimTimestamp);
+    }
+
+    /// @notice Claim liquidity assets from a previous withdrawal request after the claim delay has passed
+    /// @param requestId The index of the withdrawal request
+    /// @return assets The amount of liquidity assets that were transferred to the redeemer
+    function claimRedeem(uint256 requestId) external returns (uint256 assets) {
+        // Update the ARM's withdrawal queue's claimable amount
+        _updateWithdrawalQueueLiquidity();
+
+        // Load the structs from storage into memory
+        WithdrawalRequest memory request = withdrawalRequests[requestId];
+
+        require(request.claimTimestamp <= block.timestamp, "Claim delay not met");
+        // If there isn't enough reserved liquidity in the queue to claim
+        require(request.queued <= withdrawsClaimable, "Queue pending liquidity");
+        require(request.withdrawer == msg.sender, "Not requester");
+        require(request.claimed == false, "Already claimed");
+
+        // Store the request as claimed
+        withdrawalRequests[requestId].claimed = true;
+        // Store the updated claimed amount
+        withdrawsClaimed += request.assets;
+
+        assets = request.assets;
+
+        emit RedeemClaimed(msg.sender, requestId, assets);
+
+        // transfer the liquidity asset to the withdrawer
+        IERC20(liquidityAsset).transfer(msg.sender, assets);
+    }
+
+    /// @dev Updates the claimable amount in the ARM's withdrawal queue.
+    /// That's the amount that is used to check if a request can be claimed or not.
+    function _updateWithdrawalQueueLiquidity() internal {
```

Comment:
👽 I think we can completely eliminate both the `_updateWithdrawalQueueLiquidity()` and the `withdrawsClaimable` storage slot. These are only read or written inside the `claimRedeem()` function. In essence, it's just a question of if we can redeem right now.

If we remove these, perhaps they could be replaced by a method that looks something like this?

```solidity
    function withdrawsClaimable() public view returns (uint256) {
        return withdrawsClaimed + IERC20(liquidityAsset).balanceOf(address(this));
    }
```